spec/services/discord/sync_roles_service_spec.rb

```diff
@@ -0,0 +1,80 @@
+require "rails_helper"
+
+describe Discord::SyncRolesService do
```

Comment:
It's unclear to me whether this service spec is testing anything that `spec/system/school/discord_spec.rb` hasn't covered already. After all, `Discordrb::API::Server` is stubbed in the system spec as well. 🤔

Reply:
Yes, `spec/system/school/discord_spec.rb` actually covers more cases than this spec. Is it necessary to have a unit test even if integrated tests cover it?

Reply:
@kaisersakhi Avoiding overlap is probably the right approach, but exactly where you want to test something is a _choice_ that depends...

System tests have the advantage of testing the full stack at once, at the cost of increased brittleness and much slower speed, whereas these unit tests are much faster, but test a much smaller slice of the code that actually runs.

Reply:
I have removed the spec.